addons/web/static/src/views/pivot/pivot_view.js

```diff
@@ -38,6 +38,7 @@ export class PivotView extends Component {
             modelParams.metaData = {
                 activeMeasures: archInfo.activeMeasures,
                 additionalMeasures: this.props.additionalMeasures,
+                displayViewAsEmbed: this.props.displayViewAsEmbed,
```

Comment:
@aab-odoo , it's on me, I've driven @Brieuc-brd  to this approach :)
There is something else we use as flag to apply the right class? I'd like to avoid CSS overrides.

Reply:
I don't think there something else, but I don't think there should be. From the PivotView component point of view, he doesn't care about being embed or not. It's the same. It's only a matter of style (a single rule). What's the problem with a rule like `.o_dashboard .o_pivot { width: 100% }` ?

Reply:
> From the PivotView component point of view, he doesn't care about being embed or not

makes sense.

> What's the problem with a rule like .o_dashboard .o_pivot { width: 100% } ?

There is no problem. We're passing through all the SCSS, ideally we would like to remove everything, keeping  decorations only. Layout should be managed using classes since Bootstrap is quite powerful on that side.
In this case is a bit of a pity because we have 2 classes that suit perfectly :/

Btw, if there is nothing we can use to "catch" to embedded state we'll go with CSS 

@Brieuc-brd , sorry for the headache. I'd suggest CSS variable here, something like:

```
width: var(--o-pivot-width, auto);
```

Reply:
Hello, I'm not sure why Discuss was added on this PR at some point, but as I'm on it anyway...

@stefanorigano I think you had the right idea, don't abandon it so easily.

> From the PivotView component point of view, he doesn't care about being embed or not.

Owl 2 forced components to be context aware by removing the support from setting a class from the parent and enforcing that child set it themselves if necessary.

It's actually a good thing, and overrides are bad because they are implicit action from distance, rather than clearly defining what the component is capable of in a single place.

If we conceptually allow the style from a component to be defined from elsewhere, it means every dev needs to know 100% of the CSS and 100% of the classes that are put in any style file or template in all Odoo to know the impact of a change in any template or any CSS file. It's definitely not something desirable in terms of maintenance.

In this case the issue is a bit less glaring in terms of "hard to track" as the class is named and it's a top level element, but the same principle applied to deep sub-elements with more complex override rules would lead to really impossible understanding like before.

Not only that, but as you said Stefano the CSS override approach prevents from using existing classes and forces to write custom CSS for them, which goes against one of the goal of the current style refactoring to make the CSS more lightweight and faster.

Now don't get me wrong I don't want to enforce this idea in modules that others are maintaining and I will leave it at that, but I figure it would be a good idea to at least give it a good thought before deciding.

Reply:
> Owl 2 forced components to be context aware by removing the support from setting a class from the parent and enforcing that child set it themselves if necessary.

It's not really different imho. All the Component can decide is if and where it applies the class received in props. It doesn't prevent the parent component to give a className that will apply different rules (it's probably what will almost always happen, actually).

> It's actually a good thing, and overrides are bad because they are implicit action from distance, rather than clearly defining what the component is capable of in a single place.

I agree with that principle, but it's not compatible with odoo's modularity. When you write a framework component, you can never know all places it will be used, nor all overrides/extensions there will be. It's a pain, I would prefer this is not the case, but it is.

It's good to have guidelines and best practices, but I think that simplicity is more important. In this case, I think it's simpler to override the width in scss than to add a props (which doesn't scale good actually).

Reply:
Hello @seb-odoo , thanks for sharing!

Yes, when layouting CSS variables are better than overrides, and "toggleable" utility-classes are better that everything by a maintenance/performance point of view.

We're relying on one of the hugest CSS frameworks out there, layouting using raw SCSS shouldn't be necessary... actually we would like to avoid raw SCSS at all except if/when we are forced to. 

Having said that, I do understand if @aab-odoo wants to keep components tight and slim, it's important too.
There is a tradeoff though... this little override will not cause any trouble, but do we want to consider CSS overrides as the norm **for all** these circumstances? I'm wondering if can't we find a better way